api/v1alpha2/mysqlaaduser_types.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// MySQLAADUserSpec defines the desired state of MySQLAADUser
+type MySQLAADUserSpec struct {
+	// +kubebuilder:validation:MinLength:1
+	// +kubebuilder:validation:Required
+	Server string `json:"server"`
+
+	// +kubebuilder:validation:Pattern=^[-\w\._\(\)]+$
+	// +kubebuilder:validation:MinLength:1
+	// +kubebuilder:validation:Required
+	ResourceGroup string `json:"resourceGroup"`
+
+	// The server-level roles assigned to the user.
+	Roles []string `json:"roles"`
+
+	// The database-level roles assigned to the user (keyed by
+	// database name).
+	DatabaseRoles map[string][]string `json:"databaseRoles,omitempty"`
+
+	// AAD ID is the ID of the user in Azure Active Directory.
```

Comment:
minor: I think often we have the required stuff ordered to the top of the file.

Reply:
I've done this change in the v1alpha1 file too - I don't think there are any backwards compatibility concerns with that, right?

Reply:
Oh actually in that case Roles was required so I haven't changed it.